operator/pkg/ciliumendpointslice/endpointslice.go

```diff
@@ -62,17 +65,56 @@ func (c *Controller) initializeQueue() {
 		workqueue.TypedRateLimitingQueueConfig[CESKey]{Name: "cilium_endpoint_slice"})
 }
 
-func (c *Controller) onEndpointUpdate(cep *cilium_api_v2.CiliumEndpoint) {
-	if cep.Status.Networking == nil || cep.Status.Identity == nil || cep.GetName() == "" || cep.Namespace == "" {
+// On Pod Update, verify all the necessary fields are set.
+func (c *Controller) onPodUpdate(pod *slim_corev1.Pod) {
+	if pod.GetName() == "" || pod.Namespace == "" {
 		return
 	}
-	touchedCESs := c.manager.UpdateCEPMapping(k8s.ConvertCEPToCoreCEP(cep), cep.Namespace)
+
+	if pod.Spec.HostNetwork { // no CEP for host networking pods
+		return
+	}
+
+	_, err := GetPodEndpointNetworking(pod)
+	if err != nil {
+		c.logger.Info("could not get endpointnetworking for pod",
+			logfields.K8sPodName, pod.Name,
+			logfields.Error, err)
+		return
+	}
+
+	node, err := c.reconciler.getNodeNameForPod(pod)
+	if err != nil {
+		c.logger.Info("could not get node name for pod",
+			logfields.K8sPodName, pod.Name,
+			logfields.Error, err)
+		return
+	}
+
+	pCid, err := c.reconciler.getPodIdentity(pod)
+	if err != nil {
+		// The pod does not have a CID created for it yet. Pods will not be updated when the CID associated with
+		// them is created. We preserve information about pods and their labels, so
+		// that when the CID is created, the created CES can be reconciled with the apiserver.
+		k8sLabels, errLabels := ciliumidentity.GetRelevantLabelsForPod(c.logger, pod, c.reconciler.namespaceStore)
+		if errLabels != nil {
+			c.logger.Info("could not get labels for pod",
+				logfields.K8sPodName, pod.Name,
+				logfields.Error, err)
+			return
```

Comment:
I assume this should return an error to eventually retrigger a reconciliation, in case the namespace is not found (i.e., the informer is lagging behind). Otherwise, that pod will never receive a CES entry.